### PR TITLE
Create v0.28.2 RC

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** Lightning wheels are not building properly for Linux.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/wheel_macos_x86_64.yml [hunk 1]
```diff
@@ -186,7 +186,7 @@ jobs:
           regex: '.*[0-9]+.[0-9]+.[0-9]+[-_]?rc[0-9]+'
 
       - uses: actions/upload-artifact@v2
-        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
+        # if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
```
Comment (AmintorDusko) [inline .github/workflows/wheel_macos_x86_64.yml:189]: Before the Release change:
```suggestion
        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
```

File: .github/workflows/wheel_win_x86_64.yml [hunk 2]
```diff
@@ -189,7 +189,7 @@ jobs:
           regex: '.*[0-9]+.[0-9]+.[0-9]+[-_]?rc[0-9]+'
 
       - uses: actions/upload-artifact@v2
-        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
+        # if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
```
Comment (AmintorDusko) [inline .github/workflows/wheel_win_x86_64.yml:192]:
```suggestion
        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' || steps.rc_build.outputs.match != ''}}
```

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,7 +1,8 @@
 ninja
 flaky
 numpy
-git+https://github.com/PennyLaneAI/pennylane.git@master
+# git+https://github.com/PennyLaneAI/pennylane.git@master
+pennylane~=0.28
```
Comment (AmintorDusko) [inline requirements-dev.txt:5]: If merging to master, this needs to be changed:
```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

**Comment by mlxd** [comment]
This PR needs to be deleted as the branch tag conflict will not disappear without it. I have created a new branch v0.28.2_rc0 of which all the contents here are copied.

**Pull request closed without merging**